docs/general/ad-filtering/create-own-filters.md

```diff
@@ -1544,46 +1546,103 @@ We **strongly recommend** using these markers any time when you use an extended
+> - disallow `:has()` inside the pseudos accepting only compound selectors;
+> - disallow `:has()` after regular pseudo-elements.
 
-**Examples**
+> Native `:has()` pseudo-class does not allow `:has()`, `:is()`, `:where()` inside `:has()` argument to avoid increasing the `:has()` invalidation complexity ([case 1](https://bugs.chromium.org/p/chromium/issues/detail?id=669058#c54)). But ExtendedCss did not have such limitation earlier and filter lists already contain such rules, so we will not add this limitation in ExtendedCss and allow to use `:has()` inside `:has()` as it was possible before. To use it, just force ExtendedCss usage by setting `#?#`/`#$?#` rule marker.
+
+> Native implementation does not allow any usage of `:scope` inside `:has()` argument ([[1]](https://github.com/w3c/csswg-drafts/issues/7211), [[2]](https://github.com/w3c/csswg-drafts/issues/6399)). Still there some such rules in filter lists: `div:has(:scope > a)` which we will continue to support simply converting them to `div:has(> a)` as it was earlier.
 
-Selecting all `div` elements, which contain an element with the `banner` class:
+**Examples**
 
+`div:has(.banner)` selects all `div` elements, which **includes** an element with the `banner` class:
 ```html
 <!-- HTML code -->
-<div>Do not select this div</div>
-<div>Select this div<span class="banner"></span></div>
+<div>Not selected</div>
+<div>Selected
+  <span class="banner">inner element</span>
+</div>
 ```
 
-Selector:
+`div:has(> .banner)` selects all `div` elements, which **includes** an `banner` class element as a *direct child* of `div`:
+```html
+<!-- HTML code -->
+<div>Not selected</div>
+<div>Selected
+  <p class="banner">child element</p>
+</div>
 ```
-div:has(.banner)
+
+`div:has(+ .banner)` selects all `div` elements **preceding** `banner` class element which *immediately follows* the `div` and both are children of the same parent:
+```html
+<!-- HTML code -->
+<div>Not selected</div>
+<div>Selected</div>
+<p class="banner">adjacent sibling</p>
+<span>Not selected</span>
 ```
 
-Backward compatible syntax:
+`div:has(~ .banner)` selects all `div` elements **preceding** `banner` class element which *follows* the `div` but *not necessarily immediately* and both are children of the same parent:
+```html
+<!-- HTML code -->
+<div>Not selected</div>
+<div>Selected</div>
+<span>Not selected</span>
+<p class="banner">general sibling</p>
 ```
-div[-ext-has=".banner"]
+
+`div:has(span, .banner)` selects all `div` elements, which **includes both** `span` element and `banner` class element:
+```html
+<!-- HTML code -->
+<div>Not selected</div>
+<div>Selected
+  <span>child span</span>
+  <p class="banner">child .banner</p>
+</div>
 ```
 
+> [Backward compatible syntax for `:has()`](https://github.com/AdguardTeam/ExtendedCss#-old-syntax-for-pseudo-class-has) is supported but not recommended.
+
 ##### Pseudo-class `:if-not()`
```

Comment:
```suggestion
```

Reply:
it should remain in the article but marker as deprecated